enclave-tls/src/core/etls_common.c

```diff
@@ -0,0 +1,199 @@
+	log_level_str = calloc(1, log_level_len);
+	memset(log_level_str, 0, log_level_len);
+	ocall_getenv(name, log_level_str, log_level_len);
+#else
+	log_level_str = getenv(name);
+#endif
+	if (log_level_str) {
+#ifdef SGX
+		if (!strcmp(log_level_str, "debug") || !strcmp(log_level_str, "DEBUG"))
+#else
+		if (!strcasecmp(log_level_str, "debug"))
+#endif
+			return ENCLAVE_TLS_LOG_LEVEL_DEBUG;
+#ifdef SGX
+		else if (!strcmp(log_level_str, "info") || !strcmp(log_level_str, "INFO"))
+#else
+		else if (!strcasecmp(log_level_str, "info"))
+#endif
+			return ENCLAVE_TLS_LOG_LEVEL_INFO;
+#ifdef SGX
+		else if (!strcmp(log_level_str, "warn") || !strcmp(log_level_str, "WARN"))
+#else
+		else if (!strcasecmp(log_level_str, "warn"))
+#endif
+			return ENCLAVE_TLS_LOG_LEVEL_WARN;
+#ifdef SGX
+		else if (!strcmp(log_level_str, "error") || !strcmp(log_level_str, "ERROR"))
+#else
+		else if (!strcasecmp(log_level_str, "error"))
+#endif
+			return ENCLAVE_TLS_LOG_LEVEL_ERROR;
+#ifdef SGX
+		else if (!strcmp(log_level_str, "fatal") || !strcmp(log_level_str, "FATAL"))
+#else
+		else if (!strcasecmp(log_level_str, "fatal"))
+#endif
+			return ENCLAVE_TLS_LOG_LEVEL_FATAL;
+#ifdef SGX
+		else if (!strcmp(log_level_str, "off") || !strcmp(log_level_str, "OFF"))
+#else
+		else if (!strcasecmp(log_level_str, "off"))
+#endif
+			return ENCLAVE_TLS_LOG_LEVEL_NONE;
+	}
+
+	return ENCLAVE_TLS_LOG_LEVEL_DEFAULT;
+}
+
+enclave_tls_err_t etls_instance_init(const char *name,
+                                     __attribute__((unused)) const char *realpath,
+                                     __attribute__((unused)) void **handle
+                                     )
+{
+#ifdef SGX
+	if (!strcmp(name, "nullcrypto"))
+		libcrypto_wrapper_nullcrypto_init();
+	else if (!strcmp(name, "nullattester"))
+		libattester_null_init();
+	else if (!strcmp(name, "nullverifier"))
+		libverifier_null_init();
+	else if (!strcmp(name, "sgx_ecdsa")) {
+		libattester_sgx_ecdsa_init();
+		libverifier_sgx_ecdsa_init();
+        }
+	else if (!strcmp(name, "sgx_ecdsa_qve"))
+		libverifier_sgx_ecdsa_qve_init();
+	else if (!strcmp(name, "sgx_la")) {
+                libattester_sgx_la_init();
+		libverifier_sgx_la_init();
+        }
+	else if (!strcmp(name, "nulltls"))
+		libtls_wrapper_nulltls_init();
+	else if (!strcmp(name, "openssl")) {
+		libtls_wrapper_openssl_init();
+		libcrypto_wrapper_openssl_init();
+        } else
+		return ENCLAVE_TLS_ERR_NO_NAME;
+#else
+	*handle = dlopen(realpath, RTLD_LAZY);
+	if (*handle == NULL) {
+		ETLS_ERR("failed on dlopen(): %s\n", dlerror());
+		return -ENCLAVE_TLS_ERR_DLOPEN;
+	}
+#endif
+
+	return ENCLAVE_TLS_ERR_NONE;
+}
+
+ssize_t etls_write(int fd, const void *buf, size_t count)
+{
+	ssize_t rc;
+#ifdef SGX
+	int sgx_status = ocall_write(&rc, fd, buf, count);
+	if (SGX_SUCCESS != sgx_status) {
+		ETLS_ERR("sgx failed to write data, sgx status: 0x%04x\n", sgx_status);
+	}
+#else
+	rc = write(fd, buf, count);
+#endif
+
+	return rc;
+}
```

Comment:
There are two styles for conditional compilation:
```C
// style 1
static void test(void) {
# ifdef FOO
   // something
# else
  // else
# endif
  // ...
}

// style 2
# ifdef FOO
static void test(void) {
   // something
}
# else
static void test(void) {
   // something
}
# endif
```

If most of codes in a function requires conditional compilation, style 2 is preferred. Otherwise, use style 1. Above two functions should follow style 2.